heareval/tasks/pipeline.py

```diff
@@ -447,50 +418,40 @@ def requires(self):
             "metadata": self.metadata_task,
         }
 
-    def get_subsample_metadata(self):
-        metadata = self.metadata[["split", "subsample_key", "slug", "relpath"]]
-
-        if self.task_config["embedding_type"] == "scene":
-            assert metadata["subsample_key"].nunique() == len(metadata)
-
-        # Since event detection metadata will have duplicates, we de-dup
-        subsample_metadata = (
-            metadata[metadata["split"] == self.split].sort_values(by="subsample_key")
-            # Drop duplicates as the subsample key is expected to be unique
-            .drop_duplicates(subset="subsample_key", ignore_index=True)
-        )
-        return subsample_metadata
-
     def run(self):
-        subsample_metadata = self.get_subsample_metadata()
-        num_files = len(subsample_metadata)
+        split_metadata = self.metadata[self.metadata["split"] == self.split]
+        relpaths = split_metadata["relpath"].unique()
+        rng = random.Random("SubsampleSplit")
+        rng.shuffle(relpaths)
+        num_files = len(relpaths)
+
         # This might round badly for small corpora with long audio :\
-        # TODO: Issue to check for this
+        # But we aren't going to use audio that is more than a couple
+        # minutes or the timestamp embeddings will explode
         sample_duration = self.task_config["sample_duration"]
         max_files = int(MAX_TASK_DURATION_BY_SPLIT[self.split] / sample_duration)
+
         if num_files > max_files:
             print(
                 f"{num_files} audio files in corpus."
                 f"Max files to subsample: {max_files}"
             )
-            sampled_subsample_metadata = perform_metadata_subsampling(
-                subsample_metadata, max_files
-            )
-            print(
-                "Datapoints in split after resampling: "
-                f"{len(sampled_subsample_metadata)}"
-            )
-            assert perform_metadata_subsampling(
-                subsample_metadata.sample(frac=1), max_files
-            ).equals(sampled_subsample_metadata), "The subsampling is not stable"
+            subsampled_relpaths = set(relpaths[:max_files])
+            print(f"Files in split after resampling: f{len(subsampled_relpaths)}")
```

Comment:
Stay f in the string right before the {} -- would be nice to include the split name as well (already mentioned in previous PR)

Reply:
Fixed